### PR TITLE
User Model for Annotate and Highlight Tasks

### PR DESCRIPTION
**Pull request by JDDelrosario**
This PR creates the PageAnnotation model and required associations (user has many page annotations, page annotation belongs to user). It also includes one controller method under the SaasbookController that creates a new annotation or updates an existing annotation.

Stuff to note:
-Highlights and Annotations are now the same thing! We are using the Recogito.js library for both. This means that all highlights and annotations on a page are exported as a single JSON string. 
-Added schema.rb to the rubocop ignore file, because rubocop really shouldn't be messing with it
-Several new test files were added to accommodate testing with devise

### CODE REVIEW

File: app/views/saasbook/_table_of_contents.html.erb [hunk 1]
```diff
@@ -17,6 +17,7 @@
 	<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
     <div class="bd-toc-item active">
         <ul class="nav bd-sidenav">
+        <a class="login-button" href= "<%= user_github_omniauth_authorize_path %>">LOGIN</a>
```
Comment (Lucitic) [inline app/views/saasbook/_table_of_contents.html.erb]: is this a temporary login until the other part is merged? aren't we putting it on the topbar, so is this necessary to merge?

File: spec/controllers/saasbook_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      login_user
+      it 'should create a new annotation' do
+        @user = User.find_by(uid: 43_231, provider: 'github')
+        @annotation_find = @user.page_annotations.find_by(chapter: '1', section: '4')
+        expect(@annotation_find).to be_nil
+
+        post :annotate, params: { chapter: '1', section: '4', annotation: 'hello world' }
+
+        @annotation_find = @user.page_annotations.find_by(chapter: '1', section: '4')
+        expect(@annotation_find).not_to be_nil
```
Comment (Lucitic) [inline spec/controllers/saasbook_spec.rb:21]: maybe we should add a case for multiple annotations, making sure both are saved

**Review by Lucitic — APPROVED** [review]

File: app/views/saasbook/_table_of_contents.html.erb (hunk 1 above)
Comment (JDDelrosario) [reply app/views/saasbook/_table_of_contents.html.erb]: Actually that file wasn't supposed to get merged, good catch.

**Comment by JDDelrosario** [comment]
Added more/stronger tests, and also fixed the incorrectly added file

**Review by whellen0112 — APPROVED** [review]
Looks good to me!

**Pull request merged**